src/sdk/lsf-sdk.js

```diff
@@ -384,8 +390,7 @@ export class LSFWrapper {
         c.history.safeUnfreeze();
       }
     }
-
-    const first = this.annotations[0];
+    const first = this.annotations?.length ? this.annotations[0] : null;
```

Comment:
So it basically changes `undefined` to `null`? Is it required? Or `annotations` can be not defined? `?.[0]` will do the trick

Reply:
there were warnings all over about checking the index directly, index `0` was out of bounds. It was repeatedly doing this many times. The warnings all state to check length first. Something about mobx observable arrays does not like strictly checking the value directly like that.

Reply:
Its the way mobx patches observables, its doing it in a way that the object will always exist, but checking indices on the underlying array can be unsafe.

Reply:
![Screen Shot 2022-11-22 at 11 49 39 AM](https://user-images.githubusercontent.com/2242258/203385650-093da080-c79a-4a54-8a5b-b21e47d4da91.png)
This is what happens without the change I made, and it does this the same if you try and check with the optional `?.[0]`

Reply:
![Screen Shot 2022-11-22 at 11 52 35 AM](https://user-images.githubusercontent.com/2242258/203386153-a3e02b06-e041-4d32-8437-9ba9d7dbfdd5.png)
and here is the follow through on the offending line, you can see I tried this exactly

Reply:
oh well... and that's not a thing you'll comment about on every occurence...
that's cool you fixed this at least in these places